pyrelational/oracle/generic_oracle.py

```diff
@@ -16,10 +16,59 @@ class GenericOracle(ABC):
     def __init__(self):
         super(GenericOracle, self).__init__()
 
-    @abstractmethod
-    def update_dataset(data_manager: GenericDataManager, indices: List[int]) -> List[Any]:
+    def update_target_value(self, data_manager: GenericDataManager, idx: int, value: Any) -> None:
+        """Update the target value for the observation denoted by the index
+
+        :param data_manager: reference to the data_manager whose dataset we want to update
+        :param idx: index to the observation we want to update
+        :param value: value to update the observation with
+        """
+        data_manager.set_target_value(idx=idx, value=value)
+
+    def update_target_values(self, data_manager: GenericDataManager, indices: List[int], values: List[Any]) -> None:
+        """Updates the target values of the observations at the supplied indices
+
+        :param data_manager: reference to the data_manager whose dataset we want to update
+        :param indices: list of indices to observations whose target values we want to update
+        :param values: list of values which we want to assign to the corresponding observations in indices
+        """
+        for idx, val in zip(indices, values):
+            data_manager.set_target_value(idx=idx, value=val)
+
+    def update_annotations(self, data_manager: GenericDataManager, indices: List[int]) -> None:
+        """Calls upon the data_manager to update the set of labelled indices with those supplied
+        as arguments. It will move the observations associated with the supplied indices from the
+        unlabelled set to the labelled set. By default any indices supplied that are already in
+        the labelled set are untouched.
+
+        Note this does not change the target values of the indices, this is handled by a method
+        in the oracle.
+
+        :param data_manager: reference to the data_manager whose sets we are adjusting
+        :param indices: list of indices selected for labelling
+        """
+        data_manager.update_train_labels(indices)
+
+    # @abstractmethod
```

Comment:
should remove this if unnecessary.

Reply:
Why should this be removed? It is used in the `update_dataset` method in the oracle, and is no longer a method in the pipeline. I can disencourage access to it by making it private.

Reply:
because it's commented out :p. If we need it, then it should be uncommented!

Reply:
Oh I see you meant the decorator!, sorry I thought the window is referring to the `update_annotations` method. Yes the decorator is now uncommented.

Reply:
Is there a way to expand the view or get a more direct highlight of the line in question in this review mode?

Reply:
Done